Realm/RLMObjectBase.mm

```diff
@@ -670,26 +672,34 @@ - (void)addNotificationBlock:(RLMObjectNotificationCallback)block
     RLMObjectBase *obj = [realm resolveThreadSafeReference:tsr];
 
     _object = realm::Object(obj->_realm->_realm, *obj->_info->objectSchema, obj->_row);
-    _token = _object.add_notification_callback(ObjectChangeCallbackWrapper{block, obj});
+    _token = _object.add_notification_callback(ObjectChangeCallbackWrapper{block, obj}, keyPaths);
 }
 
-- (void)addNotificationBlock:(RLMObjectNotificationCallback)block object:(RLMObjectBase *)obj {
+- (void)addNotificationBlock:(RLMObjectNotificationCallback)block object:(RLMObjectBase *)obj keyPaths:(KeyPathArray)keyPaths {
     _object = realm::Object(obj->_realm->_realm, *obj->_info->objectSchema, obj->_row);
     _realm = obj->_realm;
-    _token = _object.add_notification_callback(ObjectChangeCallbackWrapper{block, obj});
+    _token = _object.add_notification_callback(ObjectChangeCallbackWrapper{block, obj}, keyPaths);
 }
 
-RLMNotificationToken *RLMObjectBaseAddNotificationBlock(RLMObjectBase *obj, dispatch_queue_t queue,
+RLMNotificationToken *RLMObjectBaseAddNotificationBlock(RLMObjectBase *obj,
+                                                        NSArray<NSString *> *keyPaths,
+                                                        dispatch_queue_t queue,
                                                         RLMObjectNotificationCallback block) {
     if (!obj->_realm) {
         @throw RLMException(@"Only objects which are managed by a Realm support change notifications");
     }
 
+    KeyPathArray keyPathArray;
+    for (NSString *keyPath in keyPaths) {
+        RLMKeyPath rlmKeyPath = RLMKeyPathFromString(obj.realm.schema, obj->_objectSchema, obj->_info, keyPath);
+        keyPathArray.push_back(rlmKeyPath);
+    }
```

Comment:
This `NSArray<NSString *>` to `KeyPathArray` conversion appears in multiple places and would be worth extracting to a function.

Reply:
Should be resolved in [`9c20a19`](https://github.com/realm/realm-cocoa/pull/7249/commits/9c20a19a0db6685b1c1198eb22677080f2d3c068)